workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/InputSpecValidator.java

```diff
@@ -0,0 +1,126 @@
+
+    @Autowired
+    private ServiceSpecSchemaValidator serviceSpecSchemaValidator;
+
+    @Autowired
+    private ProfileFileValidator profileFileValidator;
+
+    @Autowired
+    private ProfileSpecSchemaValidator profileSpecSchemaValidator;
+
+    /**
+     * Steps:
+     * FileValidator for service spec
+     * SchemaValidator - Service Spec
+     * 
+     * 1. If profile Name given:
+     *      Process - getProfileFiles - return List<ProfileFile> throws exception
+     *          ServiceSpec File to Model
+     *          Get All profile Files - 
+     *      
+     * FileValidator for profiles    
+     * SchemaValidator Profiles
+     */
+    @Override
+    public boolean validate(HyscaleCommandSpec commandSpec) throws HyscaleException {
+        if (commandSpec == null) {
+            return false;
+        }
+        List<File> serviceSpecs = commandSpec.getServiceSpecFiles();
+
+        if (isInvalid(serviceSpecFileValidator, serviceSpecs)) {
+            return false;
+        }
+
+        if (isInvalid(serviceSpecSchemaValidator, serviceSpecs)) {
+            return false;
+        }
+
+        List<File> profiles = commandSpec.getProfileFiles();
+        if (profiles == null && commandSpec.getProfileName() == null) {
+            return true;
+        }
+
+        if (profiles == null && commandSpec.getProfileName() != null) {
+            profiles = ServiceProfileUtil.getProfilesFromName(serviceSpecs, commandSpec.getProfileName());
+            commandSpec.setProfileFiles(profiles);
+        }
+
+        if (commandSpec.getProfileName() == null && isInvalid(profileFileValidator, profiles)) {
+            return false;
+        }
+
+        if (isInvalid(profileSpecSchemaValidator, profiles)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean isInvalid(Validator<File> validator, List<File> inputList) throws HyscaleException {
+        logger.debug("Running validator: {}", validator.getClass());
+        boolean isInvalid = false;
+        boolean isFailed = false;
+        StringBuilder exceptionMsg = new StringBuilder();
+        for (File serviceSpecFile : inputList) {
+            try {
+                isInvalid = validator.validate(serviceSpecFile) ? isInvalid : true;
+            } catch (HyscaleException e) {
+                isFailed = true;
+                exceptionMsg.append(e.getMessage()).append("\n");
+            }
+        }
+        if (isInvalid || isFailed) {
+            logger.error("Input invalid : {}, failed: {}, error message : {}", isInvalid, isFailed,
+                    exceptionMsg.toString());
+            WorkflowLogger.logPersistedActivities();
+
+        }
+        if (isFailed) {
+            throw new HyscaleException(ControllerErrorCodes.INPUT_VALIDATION_FAILED,
+                    ToolConstants.INVALID_INPUT_ERROR_CODE, exceptionMsg.toString());
+        }
+        return isInvalid;
+    }
```

Comment:
Can these 2 blocks get merged ?

Reply:
Also, if the file is valid , we can validate the schema then and there itself right ?
Why should we separate and iterate over again ?
Applies the same for profiles !